### PR TITLE
#artwork 40 inventory management

### PR DESCRIPTION
**Pull request by JEM94**
`sail pest --filter="Inventory*" --coverage | grep "artwork/Modules/Inventor.*\|Tests:.*"`
> Tests:    188 passed (711 assertions)
  artwork/Modules/Inventory/Http/Controller/InventoryController  69..105, 73..106 / 25.6%  
  artwork/Modules/InventoryManagement/Enums/CraftsInventoryColumnType…  100.0%  
  artwork/Modules/InventoryManagement/Exports/InventoryManagementExpo…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryC…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryF…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryG…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryI…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryI…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/CraftsInventory…  100.0%  
  artwork/Modules/InventoryManagement/Http/Controller/InventoryManage…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Category/CreateCr…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCr…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCr…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraf…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Column/DuplicateC…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Column/UpdateCraf…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Column/UpdateCraf…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Column/UpdateCraf…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Export/CreateInve…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Filter/UpdateOrCr…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Group/CreateCraft…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Group/UpdateCraft…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Group/UpdateCraft…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Item/CreateCraftI…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/Item/UpdateCraftI…  100.0%  
  artwork/Modules/InventoryManagement/Http/Requests/ItemCell/UpdateCr…  100.0%  
  artwork/Modules/InventoryManagement/Models/CraftInventoryCategory ... 100.0%  
  artwork/Modules/InventoryManagement/Models/CraftInventoryGroup ...... 100.0%  
  artwork/Modules/InventoryManagement/Models/CraftInventoryItem ....... 100.0%  
  artwork/Modules/InventoryManagement/Models/CraftInventoryItemCell ... 100.0%  
  artwork/Modules/InventoryManagement/Models/CraftsInventoryColumn .... 100.0%  
  artwork/Modules/InventoryManagement/Models/InventoryManagementUserF…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/CraftInventoryCate…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/CraftInventoryGrou…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/CraftInventoryItem…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/CraftInventoryItem…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/CraftsInventoryCol…  100.0%  
  artwork/Modules/InventoryManagement/Repositories/InventoryManagemen…  100.0%  
  artwork/Modules/InventoryManagement/Services/CraftInventoryCategory…  100.0%  
  artwork/Modules/InventoryManagement/Services/CraftInventoryGroupSer…  100.0%  
  artwork/Modules/InventoryManagement/Services/CraftInventoryItemCell…  100.0%  
  artwork/Modules/InventoryManagement/Services/CraftInventoryItemServ…  100.0%  
  artwork/Modules/InventoryManagement/Services/CraftsInventoryColumnS…  100.0%  
  artwork/Modules/InventoryManagement/Services/InventoryManagementExp…  100.0%  
  artwork/Modules/InventoryManagement/Services/InventoryManagementUse…  100.0%  
  artwork/Modules/InventoryManagement/Services/InventoryResourceCalcu…  100.0%

### CODE REVIEW

File: artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequest.php [hunk 1]
```diff
@@ -18,7 +18,9 @@ public function rules(): array
             'type' => ['required', 'array:id,value'],
             'type.*.id' => Rule::enum(CraftsInventoryColumnTypeEnum::class),
             'selectOptions' => [
-                Rule::requiredIf($this->request->all()['type']['id'] === CraftsInventoryColumnTypeEnum::SELECT),
+                Rule::requiredIf(
+                    $this->request->all()['type']['id'] === CraftsInventoryColumnTypeEnum::SELECT->value
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequest.php:22]: Braucht einen isset check weil wegen malformedRequest

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php [hunk 2]
```diff
@@ -5,32 +5,47 @@
...
     {
-        /** @var CraftInventoryCategory $craftInventoryCategory */
-        $craftInventoryCategory = CraftInventoryCategory::find($id);
+        return $this->craftInventoryCategory->newInstance($attributes);
+    }
+
+    public function getNewModelQuery(): BaseBuilder|Builder
+    {
+        /** @var BaseBuilder|Builder $builder */
+        $builder = $this->craftInventoryCategory->newModelQuery();
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php:25]:
```suggestion
        return $this->craftInventoryCategory->newModelQuery();
```

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php [hunk 3]
```diff
@@ -5,32 +5,47 @@
...
+
+        return $builder;
+    }
+
+    public function find(int $id): CraftInventoryCategory|null
+    {
+        /** @var CraftInventoryCategory|null $craftInventoryCategory */
+        $craftInventoryCategory = $this->getNewModelQuery()->find($id);
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php:33]:
```suggestion
        return $this->getNewModelQuery()->find($id);
```

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryGroupRepository.php [hunk 4]
```diff
@@ -4,29 +4,48 @@
...
+
+    public function getNewModelInstance(array $attributes = []): CraftInventoryGroup
+    {
+        return $this->craftInventoryGroup->newInstance($attributes);
+    }
+
+    public function getNewModelQuery(): BaseBuilder|Builder
+    {
+        /** @var BaseBuilder|Builder $builder */
+        $builder = $this->craftInventoryGroup->newModelQuery();
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryGroupRepository.php:25]:
```suggestion
        return $this->craftInventoryGroup->newModelQuery();
```

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryGroupRepository.php [hunk 5]
```diff
@@ -4,29 +4,48 @@
...
+
+        return $builder;
+    }
+
+    public function find(int $id): CraftInventoryGroup|null
+    {
+        /** @var CraftInventoryGroup|null $craftInventoryGroup */
+        $craftInventoryGroup = $this->getNewModelQuery()->find($id);
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryGroupRepository.php:33]:
```suggestion
        return $this->getNewModelQuery()->find($id);
```

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemRepository.php [hunk 6]
```diff
@@ -4,26 +4,46 @@
...
+
+    public function getNewModelInstance(array $attributes = []): CraftInventoryItem
+    {
+        return $this->craftInventoryItem->newInstance($attributes);
+    }
+
+    public function getNewModelQuery(): BaseBuilder|Builder
+    {
+        /** @var BaseBuilder|Builder $builder */
+        $builder = $this->craftInventoryItem->newModelQuery();
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemRepository.php:26]:
```suggestion
        return $this->craftInventoryItem->newModelQuery();
```

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php:1]: Was ist eigentlich der Grund warum hier jetzt überall `null` zurück kommen darf? Jetzt muss man immer prüfen ob man ein Object bekommen hat oder ob es null ist. Soll der Code jetzt nicht berechtigterweise(?) mit einer Exception aussteigen wenn die Daten die kommen nicht korrekt sind?

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemCellService.php [hunk 7]
```diff
@@ -41,13 +37,32 @@ public function create(
...
         $this->craftInventoryItemCellRepository->updateOrFail(
             $craftInventoryItemCell,
             [
-                'cell_value' => ($cellValue ?? '')
+                'cell_value' => $cellValue
             ]
         );
     }
+
+
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Services/CraftInventoryItemCellService.php:50]:
```suggestion

```

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php [hunk 8]
```diff
@@ -8,44 +8,38 @@
...
-        $this->craftsInventoryColumnRepository->getAllOrdered()->each(
-            function (CraftsInventoryColumn $column) use ($craftInventoryItem): void {
-                $this->craftInventoryItemCellService->create(
-                    $column->id,
-                    $craftInventoryItem->id
-                );
-            }
-        );
+        /** @var CraftsInventoryColumn $column */
+        foreach ($this->craftsInventoryColumnRepository->getAllOrdered() as $column) {
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php:37]: `->each` sollte besser für den RAM sein

File: artwork/Modules/InventoryManagement/Services/InventoryManagementExportService.php [hunk 9]
```diff
@@ -4,28 +4,30 @@
...
      * @throws InvalidArgumentException
+     * @throws Exception
      */
     public function cacheRequestData(Collection $data): string
     {
-        $token = Str::random(128);
-
         //cache forgets the item after 10 seconds, time enough to download
-        $this->cacheManager->set($token, $data, 10);
+        $this->cacheManager->set($token = $this->str->random(128), $data, 10);
```
Comment (updatedData) [inline artwork/Modules/InventoryManagement/Services/InventoryManagementExportService.php:30]: Warum nicht der Str von Laravel?

File: .gitignore [hunk 10]
```diff
@@ -25,12 +25,9 @@ webpack.mix.js
 phpstan.neon
 .DS_Store
 .phpunit.cache
-infection*
```
[the remark this answers is not in the document]
Comment (updatedData) [reply .gitignore:28]: Gesehen, dass es in den Logsordner verschoben wurde. Ist resolved

File: tests/Feature/Artwork/Modules/Availability/Services/AvailabilityConflictServiceTest.php
Comment (updatedData) [inline tests/Feature/Artwork/Modules/Availability/Services/AvailabilityConflictServiceTest.php:1]: Faustregel: Feature = Controller, Unit = Funktionale tests

File: tests/Feature/Event/EventControllerTest.php [hunk 11]
```diff
@@ -34,29 +34,6 @@
     setupCalendar($this->auth_user);
 });
 
-test('views events', function () {
```
Comment (updatedData) [inline tests/Feature/Event/EventControllerTest.php:37]: Warum ist das rausgeflogen?

File: tests/Unit/Artwork/Modules/Inventory/Http/Controller/InventoryControllerTest.php
Comment (updatedData) [inline tests/Unit/Artwork/Modules/Inventory/Http/Controller/InventoryControllerTest.php:1]: Das ist ein FeatureTest. Hier bereitest du einen Request vor, den du zum Controller schickst und danach vergleichst du ob die Response dem entspricht was du erwartest. Kein Bedarf für Mocks

File: tests/Unit/Artwork/Modules/InventoryManagement/Enums/CraftsInventoryColumnTypeEnumTest.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+                    'DATE',
+                    'CHECKBOX',
+                    'SELECT'
+                ]
+            ]
+        ];
+    }
+
+    /** @dataProvider enumTestDataProvider */
+    public function testEnums(array $expectedCases): void
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Enums/CraftsInventoryColumnTypeEnumTest.php:28]: Der Test sollte so geschrieben sein, dass er immer die gleichen Dinge testet. Im DataProvider werden die Variablen für den Test gesetzt

File: tests/Unit/Artwork/Modules/InventoryManagement/Exports/InventoryManagementExportTest.php [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+        $inventoryManagementExport->view();
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testGettersAndSetters(): void
+    {
+        $expectedCollection = SupportCollection::make(['abc', 'def', 'hij']);
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Exports/InventoryManagementExportTest.php:69]: Sowas in den DataProvider

File: tests/Unit/Artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryItemCellControllerTest.php [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->translatorMock = $this->getMockBuilder(Translator::class)
+            ->onlyMethods(['get'])
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    public function testUpdateCellValueSuccess(): void
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Http/Controller/CraftInventoryItemCellControllerTest.php:60]: Dein Success und Error Test kann man zusammenfassen

File: tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+namespace Tests\Unit\Artwork\Modules\InventoryManagement\Http\Requests\Category;
+
+use Artwork\Modules\InventoryManagement\Http\Requests\Category\UpdateCraftInventoryCategoryNameRequest;
+use PHPUnit\Framework\TestCase;
+
+class UpdateCraftInventoryCategoryNameRequestTest extends TestCase
+{
+    public function testRules(): void
+    {
+        self::assertSame(
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php:12]: In Tests lieber im `$this` context bleiben

File: tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequestTest.php [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+        self::assertTrue($rules['selectOptions'][0]->condition);
+
+        self::assertArrayHasKey('selectOptions.*', $rules);
+        self::assertSame('required|string', $rules['selectOptions.*']);
+
+        self::assertArrayHasKey('defaultOption', $rules);
+        self::assertSame('nullable|string', $rules['defaultOption']);
+
+        //test without required select option types
+        $request = new CreateCraftsInventoryColumnRequest();
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequestTest.php:72]: DataProvider nutzen

File: tests/Unit/Artwork/Modules/InventoryManagement/Models/CraftInventoryCategoryTest.php [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+use Artwork\Modules\InventoryManagement\Models\CraftInventoryCategory;
+use Artwork\Modules\InventoryManagement\Models\CraftInventoryGroup;
+use Illuminate\Database\Connection;
+use Illuminate\Database\ConnectionResolver;
+use Illuminate\Database\Query\Builder;
+use PHPUnit\Framework\TestCase;
+
+class CraftInventoryCategoryTest extends TestCase
+{
+    private readonly Builder $queryBuilderMock;
```
Comment (updatedData) [inline tests/Unit/Artwork/Modules/InventoryManagement/Models/CraftInventoryCategoryTest.php:15]: Du kannst auch bei Unittests gegen die DB testen

File: tests/Unit/Artwork/Modules/Project/ProjectSettingsServiceTest.php
Comment (updatedData) [inline tests/Unit/Artwork/Modules/Project/ProjectSettingsServiceTest.php:1]: Der Test sieht komplett anders aus als die anderen und die Kommentare wirken wie ein innerer Monolog

**Review by updatedData — CHANGES_REQUESTED** [review]
Wir sollten nochmal über Tests und die Verwendung von Mocks reden. Außerdem ist in dem PR zu viel refactoring für einen Featurebranch drin. Das macht es schwer die neuen Funktionen zu erkennen und zu validieren

File: app/Http/Controllers/EventController.php [hunk 18]
```diff
@@ -112,7 +113,7 @@ public function viewEventIndex(
                 $roomAttributeService,
                 $areaService,
                 $projectService,
-                Auth::user()->getCalendarFilter(),
```
[the remark this answers is not in the document]
Comment (JEM94) [reply app/Http/Controllers/EventController.php:115]: Der Code der dabei eingebaut wurde hat einen Fehler verursacht, ich weiß aber nicht mehr genau welcher. Das ist nicht Testbar gewesen und hat die Logik aufgebläht. Daher habe ich es auf den alten Stand reverted. Das muss wenn dann direkt so angepasst werden, das es auch langfristig nicht mehr angefasst werden muss.

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php (hunk 2 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php:25]: Das ist absichtlich so damit PHPStorm die Objekte richtig resolved und man in der Vervollständigung z.B. beim Model auch direkt alle relations hat und man nicht immer rumraten muss.

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php (hunk 3 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php:33]: Das ist absichtlich so damit PHPStorm die Objekte richtig resolved und man in der Vervollständigung z.B. beim Model auch direkt alle relations hat und man nicht immer rumraten muss. 👍

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php:1]: Sorry ich weiß nicht was du meinst. In dem Repo gibts keine null returns

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemCellService.php (hunk 7 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Services/CraftInventoryItemCellService.php:50]: Ey so nice dass du das siehst hahaha :D

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php (hunk 8 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php:37]: Lässt sich nur kompliziert testen. Performance ist hier aber 1a, da liegen nicht viele Daten hinter.

Dann muss man zwei Tests schreiben, einer der das each expected und einer mit ner richtigen Collection wo man den Callback testet. Wenn man dann noch auf n Service ausm constructor zugreift hat man den Salat.

File: artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequest.php (hunk 1 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Http/Requests/Column/CreateCraftsInventoryColumnRequest.php:22]: Ty wird angepasst

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php
Comment (updatedData) [reply artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php:1]: Sorry hab mich in der Datei vertan. Im artwork/Modules/InventoryManagement/Repositories/CraftInventoryGroupRepository.php z.B. gibt es jetzt null returns

File: tests/Unit/Artwork/Modules/Project/ProjectSettingsServiceTest.php
Comment (JEM94) [reply tests/Unit/Artwork/Modules/Project/ProjectSettingsServiceTest.php:1]: Das war ein Beispiel für die anderen.

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php (hunk 8 above)
Comment (updatedData) [reply artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php:37]: Es spricht ja nichts dagegen gegen die DB zu testen. In den meisten Fällen ist das sogar gut so, weil du so gleich sicherstellen kannst, dass es richtig in die db geschrieben und gelesen wird.

File: tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php (hunk 15 above)
Comment (JEM94) [reply tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php:12]: Gibt es da einen Grund für, wenn ja, welchen? Mein Blickpunkt ist: Am Ende wird eine statische Methode aufgerufen, ob mit self:: oder $this->

File: artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Repositories/CraftInventoryItemCellRepository.php:1]: Die eine Methode liefert null zurück weil Laravel-Model-find null liefern kann wenn es anhand der ID nichts findet.
Das find läuft intern in "first" rein. Wenn die Collection nix hat bekommt man null.

grafik

File: artwork/Modules/InventoryManagement/Services/InventoryManagementExportService.php (hunk 9 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Services/InventoryManagementExportService.php:30]: Weil man das nicht mit PHPUnit mocken kann. 

```
use DragonCode\Support\Helpers\Str;
```

Kann man glückerlichweise mocken, somit ist die Code-Coverage hier auf 100%

File: tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php (hunk 15 above)
Comment (updatedData) [reply tests/Unit/Artwork/Modules/InventoryManagement/Http/Requests/Category/UpdateCraftInventoryCategoryNameRequestTest.php:12]: Laut https://docs.phpunit.de/en/11.2/assertions.html gibt es keine "richtige" Art und weise wie man es schreibt. Mein Argument dafür ist einfach nur Konsistenz. Bei DataProvidern war es z.B. lange Zeit egal ob statisch oder nicht. Jetzt *müssen* sie `public static` sein

File: artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php (hunk 8 above)
Comment (JEM94) [reply artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php:37]: Tut es nicht, das macht man aber im Feature Test (die hier nicht geschrieben wurden). Unit Test werden üblicherweise nicht im Applikations-Kontext ausgeführt. Daher gibt es keine DB/keine Connections/keine Configs.

Unit -> Feature -> Application (-Test)

File: tests/Feature/Event/EventControllerTest.php (hunk 11 above)
Comment (JEM94) [reply tests/Feature/Event/EventControllerTest.php:37]: Die Route wird nirgends benutzt und der Test hat Fehler geschmissen

File: tests/Feature/Artwork/Modules/Availability/Services/AvailabilityConflictServiceTest.php
Comment (JEM94) [reply tests/Feature/Artwork/Modules/Availability/Services/AvailabilityConflictServiceTest.php:1]: Ich habe den Test nicht geschrieben, sondern nur verschoben.

File: tests/Unit/Artwork/Modules/Inventory/Http/Controller/InventoryControllerTest.php
Comment (JEM94) [reply tests/Unit/Artwork/Modules/Inventory/Http/Controller/InventoryControllerTest.php:1]: Nope das ist ein Unit Test. Beim Controller scheiden sich die Geister. Ich bin eh der Meinung doppelt hält besser. Also Unit Test um UUT zu verifizieren, Feature Test (die hier nicht teil von sind) um zu prüfen ob das Endergebnis richtig ist.

Ohne Unit test kannst du nie sicher sein, dass der Controller das tut was er machen soll, selbst wenn im Feature Test das Ergebnis am Ende richtig ist. Wenn dabei aber noch 10 Resourcen gelöscht werden kriegt das keiner mit (grobes Beispiel).

Wenn du im Feature Test erwartest, dass DB Ergebnis XYZ hat, dann reicht es die Controller Methode so anzupassen, dass XYZ in der DB steht. Da würde ein plain insert SQL genügen. Der Unit Test fliegt dir sofort um die Ohren.

**Pull request merged**